### DISCUSSION TITLE
[V3] wire:navigate doesn't load scripts on subsequent page (@push)

### DISCUSSION THREAD

**Discussion by Marre-86** [livewire/livewire#6149] [1. Help]
The part of my **'layouts.main'**
```
    <li>
        <a href="{{ route('activities.create') }}" wire:navigate.hover>Add a-y</a>
    </li>
    <li>
        <a href="{{ route('activities.index') }}" wire:navigate.hover>Activities</a>
    </li>
```
My **'create.blade.php'**:
```
@extends('layouts.main')

@push('scripts')
    <!-- For Tags input-->
    <script src="https://code.jquery.com/jquery-3.7.0.min.js"></script>
    <link href="https://cdn.jsdelivr.net/npm/select2@4.1.0-rc.0/dist/css/select2.min.css" rel="stylesheet" />
    <script src="https://cdn.jsdelivr.net/npm/select2@4.1.0-rc.0/dist/js/select2.min.js"></script>
@endpush

@section('content')

.............................................................

    <script>
        $(document).ready(function() {
            $('#tags').select2({
                tags: true,
                tokenSeparators: [','],
            });
        });
    </script>

@endsection
```
My **'index.blade.php'**:
```
@extends('layouts.main')
@section('content')

            <h3>Your activities</h3>
        @livewire('search-activities')

  
@endsection
```
The problem is that those scripts from **'create'** are not being loaded if I start session with **'index'** and then click on **'add activity'** (**'create'** route). Important to note, that if I remove `@livewire('search-activities')` line from **'index.blade.php'**, then scripts are loaded as expected.

**Comment by wizzymore** [comment]
Based on this:
https://livewire.laravel.com/docs/navigate#dont-rely-on-domcontentloaded

Instead of using:
```
<script>
        $(document).ready(function() {
            $('#tags').select2({
                tags: true,
                tokenSeparators: [','],
            });
        });
    </script>
```

try to use

```
<script>
        document.addEventListener('livewire:navigated', function() {
            $('#tags').select2({
                tags: true,
                tokenSeparators: [','],
            });
        });
    </script>
```

**Reply by Marre-86** [reply]
It made things worse. Now even after refreshing the page the scripts are not loading (before they were)

I want to notice once again, that if there are no some Livewire component inclusion on the first page ('**index**'), when the scripts on the subsequent page ('**create**') are loading successfully. Otherwise they are not, but if I refresh the page (F5) then they are loaded.

**Reply by dea96** [reply]
> Based on this: https://livewire.laravel.com/docs/navigate#dont-rely-on-domcontentloaded
> 
> Instead of using:
> 
> ```
> <script>
>         $(document).ready(function() {
>             $('#tags').select2({
>                 tags: true,
>                 tokenSeparators: [','],
>             });
>         });
>     </script>
> ```
> 
> try to use
> 
> ```
> <script>
>         document.addEventListener('livewire:navigated', function() {
>             $('#tags').select2({
>                 tags: true,
>                 tokenSeparators: [','],
>             });
>         });
>     </script>
> ```

thanks bro, you saved me

**Comment by joshhanley** [comment]
@Marre-86 there is an issue with stacks not working at all at the moment, see failing test PR #6135.

**Comment by Bloafer** [comment]
I know it's old, but I ended up using:
```
$(document).on('livewire:navigated', function(){
    $(document).trigger('ready');
});
```
So I didn't have to modify legacy libraries
